src/main/java/hudson/plugins/gradle/injection/MavenOptsHandler.java

```diff
@@ -102,7 +106,7 @@ private String filterMavenOpts(String mavenOpts) {
         return filtered;
     }
 
-    private boolean shouldBeKept(String systemProperty) {
-        return keys.stream().noneMatch(systemProperty::contains);
+    private boolean shouldBeKept(String systemProperty, Set<String> keepKeys) {
+        return keys.stream().filter(it -> !keepKeys.contains(it)).noneMatch(systemProperty::contains);
```

Comment:
My brain hangs when trying to figure what this does 🥲 , I guess `Sets.difference(keys, keepKeys).stream().noneMatch(systemProperty::contains);` is a bit more expressive but let's keep it that way.

Reply:
Yeah, `Sets.difference` is better, will replace

Reply:
Done